micropip/_commands/freeze.py

```diff
@@ -12,6 +12,12 @@
 
 
 def freeze() -> str:
+    return _freeze(REPODATA_PACKAGES, REPODATA_INFO)
+
+
+def _freeze(
+    repodata_packages: dict[str, dict[str, Any]], repodata_info: dict[str, str]
```

Comment:
(No action required for this PR)

For better typing and test coverage, we can think of using pyodide-lock package (which generates these repodata lock file) #88.

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
  from pyodide_lock import PackageSpec, InfoSpec
else:
  PackageSpec = ...
  InfoSpec = ...

_freeze(packages: PackageSpec, info: InfoSpec)
```